spec/models/rbac_spec.rb

```diff
@@ -168,6 +168,62 @@
         end
       end
 
+      context "tenant access strategy VMs and Templates" do
+        before do
+          @owned_template = FactoryGirl.create(:template_vmware, :tenant => @owner_tenant)
+          @child_tenant = FactoryGirl.create(:tenant, :divisible => false, :parent => @owner_tenant)
+          @child_group  = FactoryGirl.create(:miq_group, :tenant => @child_tenant)
+        end
+
+        context "searching MiqTemplate" do
+          it "can't see descendant tenant's templates" do
+            @owned_template.update_attributes(:tenant_id => @child_tenant.id, :miq_group_id => @child_group.id)
```

Comment:
Not that it was too important, but some people (myself included) prefer using more strict `!`-ending AR methods when possible, especially if we ignore the return result of a `save` call (or an `update_attributes` call for that matter). It allows to fail early if something goes wrong and detect possible bugs more easily. I think `update!` call wold be more beneficial here.